src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -118,33 +118,41 @@ public IFilteredTemplateInfo UnambiguousTemplateToUse
 
                 if (EnvironmentSettings.Host.TryGetHostParamDefault("prefs:language", out string defaultLanguage))
                 {
-                    IFilteredTemplateInfo match = null;
+                    IReadOnlyList<IFilteredTemplateInfo> languageMatchedTemplates = FindLanguageMatchedTemplates(_matchedTemplates, defaultLanguage);
 
-                    foreach(IFilteredTemplateInfo info in _matchedTemplates)
+                    if (languageMatchedTemplates.Count == 1)
                     {
-                        // ChoicesAndDescriptions is invoked as case insensitive 
-                        if (info.Info.Tags == null || 
-                            info.Info.Tags.TryGetValue("language", out ICacheTag languageTag) &&
-                            languageTag.ChoicesAndDescriptions.ContainsKey(defaultLanguage))
-                        {
-                            if (match == null)
-                            {
-                                match = info;
-                            }
-                            else
-                            {
-                                return UseSecondaryCriteriaToDisambiguateTemplateMatches();
-                            }
-                        }
+                        return _unambiguousTemplateToUse = languageMatchedTemplates.First();
                     }
-
-                    return (_unambiguousTemplateToUse = match) ?? UseSecondaryCriteriaToDisambiguateTemplateMatches();
                 }
 
                 return UseSecondaryCriteriaToDisambiguateTemplateMatches();
             }
         }
 
+        private static IReadOnlyList<IFilteredTemplateInfo> FindLanguageMatchedTemplates(IEnumerable<IFilteredTemplateInfo> listToFilter, string language)
+        {
+            List<IFilteredTemplateInfo> languageMatches = new List<IFilteredTemplateInfo>();
+
+            if (string.IsNullOrEmpty(language))
```

Comment:
Wouldn't this be the other way around? If no language is specified, everything matches?